torch/csrc/jit/runtime/profiling_record.cpp

```diff
@@ -1,12 +1,40 @@
 #include <torch/csrc/jit/runtime/profiling_record.h>
-#include <torch/csrc/jit/passes/clear_profiling.h>
-#include <torch/csrc/jit/passes/constant_propagation.h>
+#include <torch/csrc/jit/jit_log.h>
 #include <torch/csrc/jit/runtime/graph_executor.h>
 #include <torch/csrc/jit/runtime/interpreter.h>
+#include <ostream>
 
 namespace torch {
 namespace jit {
 
+
+
+bool ShapeSymbolTable::bindSymbolicShapes(
+    at::IntArrayRef new_sizes,
+    const c10::VaryingShape<c10::ShapeSymbol>& sym_shapes) {
+  if (!sym_shapes.size().has_value()) {
+    return true;
+  }
+  if (*sym_shapes.size() != new_sizes.size()) {
+    return false;
+  }
+  for (size_t i = 0; i < new_sizes.size(); i++) {
+    if (!sym_shapes[i].has_value()) {
+      continue;
+    }
+    auto symbol = *sym_shapes[i];
+    if (!isBound(symbol)) {
```

Comment:
This is in the hot path, so be careful about hash table lookups, this is doing two lookups even though it only needs to be doing 1.